src/battle.ts

```diff
@@ -370,7 +370,7 @@ class Pokemon implements PokemonDetails, PokemonHealth {
 		}
 		if (this.boosts[boostStat] > 6) this.boosts[boostStat] = 6;
 		if (this.boosts[boostStat] < -6) this.boosts[boostStat] = -6;
-		if (boostStat === 'accuracy' || boostStat === 'evasion') {
+		if ((this.side.battle.tier.includes('Stadium') || this.side.battle.gen > 2) && (boostStat === 'accuracy' || boostStat === 'evasion')) {
```

Comment:
Did you use `gen > 2` where you meant `gen > 1`?

Either way, this is a situation where it's a lot easier not to mess up if you have an intermediate variable:

```suggestion
		const isRBY = this.side.battle.gen <= 1 && !this.side.battle.tier.includes('Stadium');
		if (!isRBY && (boostStat === 'accuracy' || boostStat === 'evasion')) {
```

Reply:
Oops, I'm not sure how I missed that. Good catch.